src/HtmlUtils.js

```diff
@@ -102,10 +102,10 @@ var sanitizeHtmlParams = {
     // URL schemes we permit
     allowedSchemes: ['http', 'https', 'ftp', 'mailto'],
 
-    // DO NOT USE. sanitize-html allows all URL starting with '//'
-    // so this will always allow links to whatever scheme the
-    // host page is served over.
-    allowedSchemesByTag: {},
+    allowedSchemesByTag: {
+        img: [ 'data', 'mxc' ],
```

Comment:
Yeah, I would definitely say mxc: URLs are the way to go here: have the sanitiser turn them into working URLs and discard everything else.